### ISSUE TITLE
Documentation Updates

### ISSUE THREAD

**Issue by endigo9740** [skeletonlabs/skeleton#783]
As we prep for v1, I'd like to do one more sweeping pass through the documentation to polish it off. This will be a segmented pass that is made up of multiple tickets, including but not limited to the following:

## Required

- [x] https://github.com/skeletonlabs/skeleton/issues/355
- [x] https://github.com/skeletonlabs/skeleton/issues/736
- [x] https://github.com/skeletonlabs/skeleton/issues/978
- [x] https://github.com/skeletonlabs/skeleton/issues/981
- [x] https://github.com/skeletonlabs/skeleton/issues/977
- [x] https://github.com/skeletonlabs/skeleton/issues/1031
- [x] https://github.com/skeletonlabs/skeleton/issues/1049
- [x] https://github.com/skeletonlabs/skeleton/issues/1070
- [x] https://github.com/skeletonlabs/skeleton/issues/1058

## Nice to Have

- [x] Redesigned Search modal ([reference](https://tailwindui.com/components/application-ui/navigation/command-palettes))

> NOTE: I've dropped the blog section updates for now, let's circle back on those later.

**Comment by cailenxyvid** [comment]
Re: #9 - I'm happy to help with proof reading / editing tasks, when that time comes.

**Comment by endigo9740** [comment]
Modal docs: a11y APG link is incorrect and broken.

**Comment by Hugos68** [comment]
Re: item 9 - Would gladly work on this when needed.

**Comment by Hugos68** [comment]
Menu page title (singular) doesnt match up with navbar on the left hand side (plural)

**Comment by endigo9740** [comment]
Feature request: ability to deep link to headers per page - this will be invaluable for pointing things out for support.

**Comment by endigo9740** [comment]
Feature request: tabbed interface between preview/code for each feature (where relevant)

**Comment by endigo9740** [comment]
Feature request: we've had a few requests for flatter navigation. Not sure how we denote elements/components/utilities with that, but it's something worth considering.

**Comment by niktek** [comment]
Put in a request to Carbon Ads

**Comment by Hugos68** [comment]
App Shell: Provide information about why <svelte:window/> events cannot be listened to and how to work around it

**Comment by endigo9740** [comment]
> App Shell: Provide information about why svelte:window events cannot be listened to and how to work around it

NOTE: this may be handled by the `on:scroll` forwarded event with the component polish pass.

**Comment by iainnicol** [comment]
Scrolling to the bottom of a docs page (e.g. https://www.skeleton.dev/elements/typography) triggers a JavaScript error. In Chromium:

```      
Uncaught TypeError: Cannot read properties of undefined (reading 'id')
    at HTMLDivElement.ie (DocsShell-ec273fc2.js:1:3435)
```

In Firefox: 

```
Uncaught TypeError: A[0] is undefined
    Immutable 10
```
in Firefox.

And then from that point, the side menu will not consistently navigate between different pages. For example, click on Forms in the side menu. The page content will update, but the URL won't update, and the menu will still highlight Typography. Then click Alerts in the side menu: the URL will update, but not the content or the highlighted menu row.

**Comment by endigo9740** [comment]
@iainnicol thanks for sharing, but make sure you search for related issue before posting. I closed this message earlier today:
https://github.com/skeletonlabs/skeleton/issues/870

The issue has been resolved and a fix will come with our next deployment tomorrow.

**Comment by endigo9740** [comment]
NOTE: standardize our process for generating examples per each feature. Pull from a pool of common options (such as variant styles)

**Comment by endigo9740** [comment]
Idea: on the blog, provide a set of pinned blog posts at the top for useful guides. Implement category tag filtering.

**Comment by endigo9740** [comment]
Preview component: https://pbs.twimg.com/media/FnvqghyWAAAxs0F?format=jpg&name=large

**Comment by endigo9740** [comment]
This is something we should take into consideration for our documentation updates.
https://discord.com/channels/1003691521280856084/1070425953798467664/1070425953798467664

Providing more guidelines for use cases for each component like Google does with Material Design (ex: limit 3-7 items for the app rail)

**Comment by endigo9740** [comment]
Idea: redesign search as a command palette: https://tailwindui.com/components/application-ui/navigation/command-palettes

**Comment by szachy113** [comment]
I am eager to assist with anything you may need. Just let me know how I can be of service.

**Comment by endigo9740** [comment]
Reference: https://docusaurus.io/

**Comment by endigo9740** [comment]
Reminder: document @AdrianGonz97 PurgeCSS plugin (when ready) for use with Skeleton:
https://github.com/skeletonlabs/skeleton/issues/732

**Comment by endigo9740** [comment]
FYI I've updated the original post above to better reflect the upcoming process for these changes.

@cailenxyvid @Hugos68 @szachy113 let me know if you're still good to help with auditing and improving the copy for the doc site. We're not quite at that step yet, but it's coming up soon. Probably starting next week.

**Comment by endigo9740** [comment]
Cool, general idea is I'll need all three of you on Discord and I'll setup a dedicated channel for us to coordinate on this. I have a loose plan for how I want to tackle this, but I'll formalize this before then. And of course, timezones and other considerations will be taken into account. I don't expect this to be solved in a single day.

**Comment by endigo9740** [comment]
Update contribution guidelines, including linting, prettier, and other VS Code editor configs, such as the Script `cBase` style autosuggestion tweak:

```
{
    "tailwindCSS.experimental.classRegex": [["/\\*tw\\*/ ([^;]*);", "'([^']*)'"]],
}
```

**Comment by niktek** [comment]
Need to adjust lint rules to ignore backlash quotes in @html blocks

**Comment by IRod22** [comment]
@endigo9740, I was doing a sweep of the guides, and, so far, I found a grammatical error in the installation guide. Do you want me to open a PR now, do you want me to wait until I find more errors in the docs, or do you want me to wait until most of the subtickets are completed? I don't want to cause any merge conflicts, which is why I'm asking. If you want me to go ahead and open a PR, I'll need to know which commit to branch off of to avoid interfering with other PRs. I'll also need to know what directory the guides are in; I tried looking in the `src/docs` folder, but I think the guides are very deep in the docs directory.

**Comment by endigo9740** [comment]
@IRod22 were these issues on the dev or production branches? We're about to drop a whole slew of updates to those sections in tomorrow's release. Preview here:

https://dev.skeleton.dev/docs/introduction

I'd say hold off until we put out a call to arms. I have two other updates I want to get in place before other folks start touching the doc pages, otherwise we might end up duplicating effort. 

1. The doc shell update - this may potentially drastically alterer the page layout for more "inner" pages
2. The new "Previewer" component - which will allow users to quickly toggle between preview/code

I have a basic proof of concept of number two on the [Code Blocks page here](https://dev.skeleton.dev/utilities/codeblocks).

**Comment by endigo9740** [comment]
@cailenxyvid @Hugos68 @szachy113 @IRod22 FYI I've created a new `#docs-overhaul` channel on Discord. I highly encourage you guys join and ready my notes there. This details the status and plan for handling the upcoming doc audit and review from volunteers like yourself. Keep an eye on that channel this week and I'll let you know when we're ready go go.

Also please post a quick "hey I'm X person on GitHub" so I can keep everyone straight please heh. Thanks!

**Comment by IRod22** [comment]
@endigo9740, I'm sorry to say this, but I'm not on Discord; I'm trying to limit my social media footprint right now, and I'm a bit concerned about Discord's safety and security practices...

**Comment by endigo9740** [comment]
Sorry, unfortunately that's our primary place for real-time collaboration. There's zero chance I'm going to be able to replicate all information from there to here. I anticipate there will be too much. I might suggest creating a throwaway account to get you by for now, otherwise we appreciate we appreciate the offer. There's always a chance something slips through and we would welcome addition eyes in the future though.

**Comment by IRod22** [comment]
No, I'm sorry. I'll just be an RC tester.

**Comment by endigo9740** [comment]
Error on the Toast docs - `present` has been replaced with `background`

![image](https://user-images.githubusercontent.com/1509726/220531947-c539d513-73ac-4c26-80da-c505a0c46503.png)

**Comment by Sarenor** [comment]
> That brings me probably to the main problem of my confusion and why I suggested the enhancement: A lot of documents talk about the color design tokens (too much noise) and in different ways:
> 
> Theme Generator
> https://www.skeleton.dev/docs/generator
> Styling
> https://www.skeleton.dev/docs/styling
> Colors
> https://www.skeleton.dev/docs/colors
> Design Tokens
> https://www.skeleton.dev/elements/tokens
> Components
> https://www.skeleton.dev/elements/buttons
> https://www.skeleton.dev/elements/cards
> https://www.skeleton.dev/elements/alerts
> 
> Streamlining this into a "80% of your use cases" and "The full flexibility and extensibility" section could eventually help to navigate a new user through the docs while keeping an easy-to-take learning curve.

See also: https://github.com/skeletonlabs/skeleton/issues/1024

**Comment by endigo9740** [comment]
It looks like we have several broken links in our page footer - those will need to be addressed

**Comment by Hugos68** [comment]
> FYI I've updated the original post above to better reflect the upcoming process for these changes.
> 
> @cailenxyvid @Hugos68 @szachy113 let me know if you're still good to help with auditing and improving the copy for the doc site. We're not quite at that step yet, but it's coming up soon. Probably starting next week.

Missed this, Yeah I'm still down to help with cleaning up the docs 👍

**Comment by endigo9740** [comment]
Awesome, the ETA to begin is probably Monday or so next week (US time)

**Comment by endigo9740** [comment]
Type on quickstart: https://github.com/skeletonlabs/skeleton/issues/1037

**Comment by endigo9740** [comment]
Conic Gradient: let's add some explanation for creating reactive UI for these. It requires using a spread operate to create a new array. We can reference the official Svelte tutorial on this topic here:
https://svelte.dev/tutorial/updating-arrays-and-objects

**Comment by endigo9740** [comment]
Light Switch; "let's make it more clear that three options are presented on the page and they should not be used in parallel:
1. Lightswitch - plug and play, drop it in and you're done
2. OS Preference - requires a quick import and one step
3. Create your own Lightswtich - requires creating a custom component and importing the utility methods provided
https://www.skeleton.dev/utilities/lightswitches

**Comment by endigo9740** [comment]
Popups: let's ensure the setup instruction is clear. There's been some confusion on the order of imports and how to replicate Menu/Tooltip functionality.

**Comment by jhiney** [comment]
Not sure if it has been mentioned yet but the link to the contribution guidelines in the readme goes to a 404 page (/docs/contributions) rather than the actual page (/docs/contributing).

**Comment by endigo9740** [comment]
Great catch @jhiney. I'll add that to my list!

**Comment by endigo9740** [comment]
The text audit and homepage redesign issues have now been marked Closed.
